### PR TITLE
DBZ-7570/#133: add workaround using SO_LINGER with 0 timeout

### PR DESCRIPTION
**Pull request by acristu**

### CODE REVIEW

**Comment by acristu** [comment]
current workaround inspired from https://bugs.openjdk.org/browse/JDK-8241239?focusedId=14646711&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-14646711

File: pom.xml [hunk 1]
```diff
@@ -4,7 +4,7 @@
 
     <groupId>com.zendesk</groupId>
     <artifactId>mysql-binlog-connector-java</artifactId>
-    <version>0.29.1</version>
+    <version>0.29.2</version>
```
Comment (Naros) [inline pom.xml:7]: Hi, let's remove this change in the commit.  @osheroff will decide what the version should be based on this and any other changes that end up merged for the release.  It's also just cleaner to leave that task for the release process.

**Review by Naros — CHANGES_REQUESTED** [review]
Thanks for the PR @acristu, I've left one comment.

File: pom.xml (hunk 1 above)
Comment (acristu) [reply pom.xml]: done

**Review by Naros — APPROVED** [review]
@osheroff, this LGTM as it's opt-in.  Could you ping me when we have a public build of the binlog available?  Thanks!

**Comment by osheroff** [comment]
@Naros go ahead and make the pom.xml change + the CHANGELOG entry.  I'll merge and release.  I'd also like to just set you up with the credentials necessary to do releases of this repo yourself, although I can never remember the incantations required to give you access there...

**Comment by Tang8330** [comment]
@osheroff Could you publish a new version? [Latest version](https://central.sonatype.com/search?q=g:com.zendesk%20%20a:mysql-binlog-connector-java&smo=true) is still `0.29.1`

**Comment by Naros** [comment]
> I'd also like to just set you up with the credentials necessary to do releases of this repo yourself, although I can never remember the incantations required to give you access there...

Whenever you do, just drop me a note to the email in my GitHub profile @osheroff .

**Comment by osheroff** [comment]
ok published after some shenannigans with sonatype.  @Naros I spoke too soon, I don't think I can grant you access to publish under com.zendesk (I don't work there anymore).  I guess we could cut the publishing over to redhat if you want to grant me access to publish the artifact, or... eh

**Comment by Naros** [comment]
Did the release go through?  I only see 0.29.1 still on Maven Central @osheroff .

**Comment by Naros** [comment]
@osheroff would you be open to the idea of we fork this under the debezium umbrella and cut releases with our automated release processes, since this is a critical component for two of our connectors?

**Comment by osheroff** [comment]
Yeah that’s fine by me.On Jul 10, 2024, at 16:50, Chris Cranford ***@***.***> wrote:﻿
@osheroff would you be open to the idea of we fork this under the debezium umbrella and cut releases with our automated release processes, since this is a critical component for two of our connectors?

**Pull request merged**